### DISCUSSION TITLE
Flux no longer reconciles cluster, complains about gitlab-admin v1beta1 api version

### DISCUSSION THREAD

**Discussion by evanrich** [fluxcd/flux2#2175] [Q&A] [answered]
Given the following ClusterRoleBinding:
```
kubectl get clusterrolebinding gitlab-admin -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRoleBinding","metadata":{"annotations":{"kustomize.toolkit.fluxcd.io/checksum":"08460cd08b4fa485ed68582eba9b1addf83f4840"},"labels":{"kustomize.toolkit.fluxcd.io/name":"flux-system","kustomize.toolkit.fluxcd.io/namespace":"flux-system"},"name":"gitlab-admin"},"roleRef":{"apiGroup":"rbac.authorization.k8s.io","kind":"ClusterRole","name":"cluster-admin"},"subjects":[{"kind":"ServiceAccount","name":"gitlab-admin","namespace":"kube-system"}]}
    kustomize.toolkit.fluxcd.io/checksum: 08460cd08b4fa485ed68582eba9b1addf83f4840
  creationTimestamp: "2021-12-03T18:55:39Z"
  labels:
    kustomize.toolkit.fluxcd.io/name: flux-system
    kustomize.toolkit.fluxcd.io/namespace: flux-system
  name: gitlab-admin
  resourceVersion: "112164781"
  uid: 3f26ba03-5710-45d5-9cd7-e6ce48649c8c
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: gitlab-admin
  namespace: kube-system

```

I am getting the following after updating k8s to 1.22.2 and flux 0.24.0:
```
 flux get kustomizations --watch
NAME            READY   MESSAGE                                                                                                                                         REVISION                                           SUSPENDED
flux-system     False   ClusterRoleBinding/gitlab-admin dry-run failed, error: no matches for kind "ClusterRoleBinding" in version "rbac.authorization.k8s.io/v1beta1"  master/bd16572bd961e1064914bdac473f1ffa82529707    False
```
the CRB DID say v1beta1, but I edited it (including deleting and re-adding), restarted the kustomization controller in flux, and still get this message.  I've tried doing a `kubectl get clusterrolebindings -A |grep -i v1beta1` and nothing shows up now, so everything should be using the correct api version.     Flux correctly updates my github repo, it just won't reconcile the pods in the cluster.

gitlab-admin is used for gitlab, I've tried deleting this CRB completely, pushing a change to the repo, etc, and flux still complains that this resource exists.

**Comment by hiddeco** [comment, marked as answer]
When you say you "edited it (including deleting and re-adding)", it sounds like you made modifications to the live cluster state. The error that is being returned however comes from a resource in the `sourceRef`, which likely still contains a `v1beta1` reference (in e.g. Git).

**Reply by evanrich** [reply]
@hiddeco  sorry to trouble, this ended up being an issue pushing to github.  @kingdonb  pointed me in the right direction, but for some reason my changes weren't going up, so i edited the resource in github directly, and now flux reconciles correctly.   Sorry for being dense =(
